From 60715769cd5368a63b2bdc2e6d6bb6df0030a54d Mon Sep 17 00:00:00 2001
From: Peter Seiderer <ps.report@gmx.net>
Date: Thu, 31 Mar 2022 21:44:22 +0200
Subject: [PATCH] preview: fix egl_preview compile (conversion from Window to
 EGLNativeWindowType)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fixes:

  .../build/libcamera-apps-2d1009e3badcc8047361ff81149ad6cba3b911b5/preview/egl_preview.cpp:329:69: error: invalid conversion from ‘Window’ {aka ‘long unsigned int’} to ‘EGLNativeWindowType’ {aka ‘fbdev_window*’} [-fpermissive]
    329 |         egl_surface_ = eglCreateWindowSurface(egl_display_, config, window_, NULL);
        |                                                                     ^~~~~~~
        |                                                                     |
        |                                                                     Window {aka long unsigned int}

[Upstream: https://github.com/raspberrypi/libcamera-apps/pull/283]
Signed-off-by: Peter Seiderer <ps.report@gmx.net>
---
 preview/egl_preview.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/preview/egl_preview.cpp b/preview/egl_preview.cpp
index b936e57..593583d 100644
--- a/preview/egl_preview.cpp
+++ b/preview/egl_preview.cpp
@@ -326,7 +326,7 @@ void EglPreview::makeWindow(char const *name)
 	wm_delete_window_ = XInternAtom(display_, "WM_DELETE_WINDOW", False);
 	XSetWMProtocols(display_, window_, &wm_delete_window_, 1);
 
-	egl_surface_ = eglCreateWindowSurface(egl_display_, config, window_, NULL);
+	egl_surface_ = eglCreateWindowSurface(egl_display_, config, reinterpret_cast<EGLNativeWindowType>(window_), NULL);
 	if (!egl_surface_)
 		throw std::runtime_error("eglCreateWindowSurface failed");
 
-- 
2.35.1

